ComparisonWithOrleans.md

```diff
@@ -0,0 +1,203 @@
+
+#### Promises
+
+  * Orleans relies upon the async–await language feature to formulate continuations that are invoked asynchronously. This means that the user need only ensure that methods return Promises and can then write code as if it was synchronous.
+  
+  * Instead of trying to offer a familiar (synchronous) user experience Akka makes handling asynchronous results explicit. In the case of the `ask`-pattern—which is analogous to how every invocation works in Orleans—the user should transform the Future explicitly or send its value to an Actor by way of the `pipeTo` pattern.
+
+We found that making asynchronous code look synchronous invites many subtle possibilities of making mistakes, either concerning data consistency (i.e. transparently interleaving different invocations by way of their scheduled closures) or liveness (by blocking a Grain from processing any message until some response has been received). The nature of distributed systems is that messages are sent one-way and with at-most-once semantics, that is what physics gives us, and suggesting that there will always be a reply often turns out to be a pitfall, especially for unexperienced users.
+
+#### Turns
+
+  * In Orleans a Grain is scheduled in terms of non-overlapping *turns*, meaning that no two threads run that Grain’s code concurrently at any given time. This is also true for continuations formed via the usage of async–await.
+  
+  * Akka schedules Actors similarly, upon each assignment to a (single) Thread an Actor may process up to a configurable limit of messages and it is guaranteed that an Actor is running on at most one Thread at any given time. While it is possible to specify an ExecutionContext that runs continuations within the Actor’s schedule, the default executors do not respect this constraint and will typically run Future continuations concurrently with the Actor that scheduled them.
+  
+#### Persistence
+
+  * Orleans provides persistence based on snapshots that are managed by the Grain explicitly. This maps well to a database-backed model where each Grain is responsible for a conceptual “row”.
+  
+  * Akka Persistence is based upon event-sourcing, which means that the PersistentActor emits events that represent state changes to be applied. Snapshots may be used as an optimization to shorten recovery time.
+
+In both cases the persistence provider is configurable and users may choose to persist Actor state independently by other means as well. The biggest difference between snapshots and event-sourcing is that the event history of a PersistentActor contains valuable information that is lost when persisting only snapshots: each event has a well-defined meaning in the business domain and thus the event log retains its meaning as the internal representation of the Actor’s state changes over time while snapshots will become invalid and cannot be mined for additional insights later.
```

Comment:
This is a great point and we agree on the usefulness of event source persistence. This is another important extensibility point to Orleans, which we absolutely welcome as part of the 2nd principle I outlined in the beginning.
We are very actively in the process of adding such a support, with the help and in a full coordination with our open source community: https://github.com/dotnet/orleans/issues/343
Expect this to be implemented very soon.